lib/services/services.c

```diff
@@ -1079,3 +1080,82 @@ resources_list_agents(const char *standard, const char *provider)
 
     return NULL;
 }
+
+gboolean
+resources_valid_tuple(const char *standard, const char *provider, const char *agent)
```

Comment:
I'm thinking this should be named resources_agent_exists(), since technically an agent doesn't have to exist on the local host in order for the tuple to be valid for the cluster (users can use location constraints to keep certain resources only on nodes that have the software installed, for example). I would expect a resources_valid_tuple() fimctopm to just check that standard and agent are not NULL, and provider is NULL or not according to standard .